src/analyze/BuildSystem.re

```diff
@@ -130,32 +130,42 @@ let detect = (rootPath, bsconfig) => {
 let getEsyCompiledBase = (root) => {
   let env = Unix.environment()->Array.to_list;
 
-  switch(Utils.getEnvVar(~env, "cur__original_root"), Utils.getEnvVar(~env, "cur__target_dir")) {
-  | (Some(projectRoot), Some(targetDir)) => Ok(Files.relpath(projectRoot, targetDir))
-  | (_, _) =>
-    switch (Commands.execResult("esy command-env --json")) {
-    | Ok(commandEnv) =>
-      switch (Json.parse(commandEnv)) {
-      | exception (Failure(message)) =>
-        Log.log("Json response");
-        Log.log(commandEnv);
-        Error("Couldn't find Esy target directory (invalid json response: parse fail): " ++ message);
-      | exception exn =>
-        Log.log(commandEnv);
-        Error("Couldn't find Esy target directory (invalid json response) " ++ Printexc.to_string(exn));
-      | json =>
-        Json.Infix.(
-          switch (
-            Json.get("cur__original_root", json) |?> Json.string,
-            Json.get("cur__target_dir", json) |?> Json.string,
-          ) {
-          | (Some(projectRoot), Some(targetDir)) => Ok(Files.relpath(projectRoot, targetDir))
-          | _ => Error("Couldn't find Esy target directory (missing json entries)")
-          }
-        )
+  let correctSlashesOnWindows = (p) => {
+      if (Sys.win32) {
+          let slashRegex = Str.regexp("/");
+          Str.global_replace(slashRegex, "\\\\", p);
+      } else {
+          p
       }
-    | err => err
+  };
+
+  let prevCwd = Unix.getcwd();
```

Comment:
Need to clean this up - I think I'll just add a `cwd` parameter to `execResult` (and needs to be more robust in case the command fails)

Reply:
I'm not sure about this code path. Can't we encapsulate this behavior inside `Commands.execResult` for example?

Reply:
Cleaned this up in [0bed757](https://github.com/jaredly/reason-language-server/pull/199/commits/0bed757184f9ced4c1fed4ee6eed164a936bbb31) - we had similiar logic in `execFull` so I factored it out to a `withCwd` function and reused it in both places.